Host allocation sharing via set final data - construction->destruction (#502)
KhronosGroup/SYCL-Docs: adoc/chapters/programming_interface.adoc

However, optionally the [code]#buffer::set_final_data()# can be set to a
[code]#std::weak_ptr# to enable copying data back, to another host memory

> address that is going to be valid after buffer destruction.

Reviewer: If we want to commit this as an interim improvement, let's change this to something like:

> ... that will be valid when the buffer is destroyed.

Saying "after buffer destruction" is wrong because the pointer must be valid before the buffer destructor is invoked.

Author: The latest commit makes this change, so I think the conversation is resolved (but @gmlueck should confirm)

Reviewer: Thanks, @mkinsner

> address that is going to be valid after buffer destruction.

Author: ```suggestion
address that will be valid when the buffer is destroyed.
```
